src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,321 @@
+
+  /** When `true`, marks this component as required in form. */
+  @Prop({ reflect: true }) required: boolean;
+
+  /** The id of the form `textarea` is associated with. */
+  @Prop({ reflect: true }) form: string;
+
+  /** The label of the component */
+  @Prop({ reflect: true }) label: string;
+
+  /**
+   * When `true`, the component will not be visible.
+   */
+  @Prop({ reflect: true }) hidden = false;
+
+  /** When true, the `textarea` will be marked as invalid. */
+  @Prop({ reflect: true }) invalid = false;
+
+  @Watch("disabled")
+  disabledHandler(value: boolean): void {
+    if (value) {
+      this.disableSlottedElements(value);
+    }
+  }
+
+  @Prop() numberingSystem?: NumberingSystem;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Emitters
+  //
+  //--------------------------------------------------------------------------
+  /**
+   * Fires each time a new `value` is typed.
+   */
+  @Event({ cancelable: true }) calciteTextareaInput: EventEmitter<void>;
+
+  /**
+   * Fires each time a new `value` is typed and committed.
+   */
+  @Event({ cancelable: true }) calciteTextareaChange: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback(): void {
+    connectLabel(this);
+    connectForm(this);
+    if (this.disabled) {
+      this.disableSlottedElements(this.disabled);
+    }
+    connectLocalized(this);
+  }
+
+  disconnectedCallback(): void {
+    disconnectLabel(this);
+    disconnectForm(this);
+    disconnectLocalized(this);
+    this.resizeObserver?.disconnect();
+  }
+
+  render(): VNode {
+    return (
+      <Host>
+        <textarea
+          aria-disabled={this.disabled}
+          aria-invalid={this.invalid}
+          aria-label={getLabelText(this)}
+          autofocus={this.autofocus}
+          class={{
+            textarea: true,
+            [CSS.resizeDisabled]: this.resizeDisabled,
+            [CSS.resizeDisabledX]: this.horizantalResizeDisabled,
+            [CSS.resizeDisabledY]: this.verticalResizeDisabled,
+            [CSS.readonly]: this.readonly,
+            [CSS.textareaInvalid]: this.invalid,
+            [CSS.footerSlotted]: !!this.renderFooterLeading() && !!this.renderFooterTrailing()
+          }}
+          cols={this.cols}
+          disabled={this.disabled}
+          form={this.form}
+          hidden={this.hidden}
+          name={this.name}
+          onChange={this.handleChange}
+          onInput={this.handleInput}
+          placeholder={this.placeholder}
+          readonly={this.readonly}
+          ref={this.setTextareaEl}
+          required={this.required}
+          rows={this.rows}
+          value={this.value}
+          wrap={this.wrap}
+        />
+        <slot />
+        {this.footer && (
+          <footer
```

Comment:
I think we might want to add a `key` on this VNode since its conditional

Reply:
i dont think so, we only have one `footer` element here.